From 746f34946ddf5b874c22be4bf45c0baa852ebc7e Mon Sep 17 00:00:00 2001
Message-Id: <746f34946ddf5b874c22be4bf45c0baa852ebc7e.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Mon, 10 Nov 2014 19:53:41 +0530
Subject: [PATCH 42/75] MUST_REBASE [VPG]: adf/intel/vlv: Move "vlv_dc_config"
 structure to header file

By Moving "vlv_dc_config" structure to vlv_dc_config.h, we enable
config structure for other part of code to use.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I16662cef71bdcf2ed31b06006a8f9d4efa8319c1
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c   |   20 ----------------
 .../adf/intel/include/core/vlv/vlv_dc_config.h     |   25 ++++++++++++++++++++
 2 files changed, 25 insertions(+), 20 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index 6a69b0a..a11d747 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -21,26 +21,6 @@
 
 #define VLV_ID(pipe, plane) ((pipe * VLV_MAX_PLANES) + plane)
 
-struct vlv_dc_config {
-	struct intel_dc_config base;
-	struct vlv_disp {
-		struct vlv_pri_plane pplane;
-		struct vlv_sp_plane splane[2];
-		enum intel_pipe_type type;
-		union {
-			struct dsi_pipe dsi;
-
-			/* later we will have hdmi pipe */
-		} pipe;
-	} vdisp[2];
-};
-
-static inline struct vlv_dc_config *to_vlv_dc_config(
-	struct intel_dc_config *config)
-{
-	return container_of(config, struct vlv_dc_config, base);
-}
-
 static const struct intel_dc_attachment vlv_allowed_attachments[] = {
 	{
 		.pipe_id = PIPE_A,
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
index 8c085b6..f0f7602 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dc_config.h
@@ -19,6 +19,10 @@
 #include "core/vlv/vlv_dc_regs.h"
 #include <drm/i915_drm.h>
 #include <drm/i915_adf.h>
+#include <core/common/dsi/dsi_pipe.h>
+#include <core/vlv/vlv_dc_config.h>
+#include <core/vlv/vlv_pri_plane.h>
+#include <core/vlv/vlv_sp_plane.h>
 
 #define VLV_N_PLANES	6
 #define VLV_N_PIPES	2
@@ -93,6 +97,27 @@ static inline void vlv_cck_write(u32 reg, u32 val)
 			       reg, &val);
 }
 
+struct vlv_dc_config {
+	struct intel_dc_config base;
+	struct vlv_disp {
+		struct vlv_pri_plane pplane;
+		struct vlv_sp_plane splane[2];
+		enum intel_pipe_type type;
+		union {
+			struct dsi_pipe dsi;
+
+			/* later we will have hdmi pipe */
+		} pipe;
+
+	} vdisp[2];
+};
+
+static inline struct vlv_dc_config *to_vlv_dc_config(
+	struct intel_dc_config *config)
+{
+	return container_of(config, struct vlv_dc_config, base);
+}
+
 bool vlv_intf_screen_connected(struct intel_pipe *pipe);
 u32 vlv_intf_vsync_counter(struct intel_pipe *pipe, u32 interval);
 extern int pipe_mode_set(struct intel_pipe *pipe,
-- 
1.7.9.5

